react_basic/src/pages/PageWrapper.jsx

```diff
@@ -0,0 +1,154 @@
+import { createContext, useEffect, useState } from 'react';
+import { Outlet, useLocation, useNavigate } from 'react-router-dom';
+import Header from '../components/Header.jsx';
+import Footer from '../components/Footer.jsx';
+import PopupMessage from '../components/PopupMessage.jsx';
+import FullscreenPopup from '../components/FullscreenPopup.jsx';
+import Advert from '../components/Advert.jsx';
+import themeService from '../services/themeService.js';
+import localStorageService from '../services/localStorageService.js';
+import urlUtils from '../utils/urlUtils.js';
+import ErrorBoundary from '../components/ErrorBoundary.jsx';
+import ErrorMsg from '../components/ErrorMsg.jsx';
+import '../styles/PageWrapper.css';
+
+const PageWrapperContext = createContext();
```

Comment:
PageWrapperContext has too many responsibilities (auth management, theme management, notifications) it would be better to split it into separate contexts: AuthContext (AuthProvider Component), ThemeContext (ThemeProvider Component) and NotificationContext (NotificationProvider Component).